internal/build/image_builder.go

```diff
@@ -180,7 +186,7 @@ func (ib *ImageBuilder) shouldUseKINDLoad(ctx context.Context, iTarg model.Image
 	// if we're using KIND and the image has a separate ref by which it's referred to
 	// in the cluster, that implies that we have a local registry in place, and should
 	// push to that instead of using KIND load.
-	if iTarg.HasDistinctClusterRef() {
+	if refs.LocalRef != refs.ClusterRef {
```

Comment:
i don't think this equality check is right? i think you need to convert to strings before comparing